services/web/server/src/simcore_service_webserver/garbage_collector_core.py

```diff
@@ -350,17 +354,19 @@ async def _remove_single_orphaned_service(
             # 1. opened project disappeared from redis?
             # 2. something bad happened when closing a project?
 
-            # TODO: logic around save_state is not ideal, but it remains
-            # with the same logic as before until it is properly refactored
+            user_id = int(interactive_service.get("user_id", -1))
+
             user_role: Optional[UserRole] = None
             try:
-                user_role = await get_user_role(
-                    app, user_id=int(interactive_service.get("user_id", -1))
-                )
+                user_role = await get_user_role(app, user_id)
             except (UserNotFoundError, ValueError):
                 user_role = None
 
-            save_state = True
+            project_uuid = currently_opened_projects_node_ids[service_uuid]
+
+            save_state = await _get_project_db_api(app).has_permission(
+                user_id, project_uuid, "write"
```

Comment:
question: what happens if you call with -1?

Reply:
I've added a case for this case, to be sure. Will not be a problem